pkg/apis/core/types.go

```diff
@@ -3209,6 +3209,18 @@ type LoadBalancerIngress struct {
 	Hostname string
 }
 
+// ServiceIPFamily represents the IP Family which the service will expose
+type ServiceIPFamily string
+
+const (
+	// Default value of ServiceIPFamily. Represents the IPFamily of the first of ServiceCIDRs  assigned to this cluster.
+	ServiceIPFamilyClusterDefault ServiceIPFamily = "ClusterDefault"
+	// IPv4Service indicates that this service will expose IPv4 IP and will select only endpoints that are IPv4
+	IPv4Service ServiceIPFamily = "IPv4Service"
```

Comment:
Name this `IPv4`? 

`serviceIPFamily: IPv4` vs `serviceIPFamily: IPv4Service`

Reply:
if you don't mind. I will leave naming until we come to an agreement (i am not saying we should keep it as is, as i said in PR docs. These are horrible names :-) ). I tried to lookup api rules for naming inspiration and couldn't find any. Maybe i missed it?. So it will be a point of coming to agreement CC @thockin

Reply:
for constants I would for with "Default", IPv4", and "IPv6"

Reply:
@lavalamp suggests NOT enumerating "default" and instead setting the default value to actual family being chosen.  I'm not sure how to get that plumbed all the way into defaulting code.

Reply:
We don't wire up the ranges all the way to validation/defaulting code. Default helps in upgrade scenario in this case.